ansible_rulebook/rule_set_runner.py

```diff
@@ -435,6 +447,46 @@ async def _call_action(
             except BaseException as e:
                 logger.error(e)
                 raise
+        elif action_plugin := self.find_action(action):
+            try:
+                await action_plugin["main"](
+                    event_log=self.event_log,
+                    inventory=inventory,
+                    hosts=hosts,
+                    variables=variables,
+                    project_data_file=self.project_data_file,
+                    source_ruleset_name=ruleset,
+                    source_ruleset_uuid=ruleset_uuid,
+                    source_rule_name=rule,
+                    source_rule_uuid=rule_uuid,
+                    rule_run_at=rule_run_at,
+                    **action_args,
+                )
+            except Exception as e:
+                logger.error("Error calling action %s, err %s", action, str(e))
+            except BaseException as e:
+                logger.error(e)
+                raise
+        elif action_plugin := find_action(*split_collection_name(action)):
+            try:
+                await action_plugin.main(
+                    event_log=self.event_log,
+                    inventory=inventory,
+                    hosts=hosts,
+                    variables=variables,
+                    project_data_file=self.project_data_file,
+                    source_ruleset_name=ruleset,
+                    source_ruleset_uuid=ruleset_uuid,
+                    source_rule_name=rule,
+                    source_rule_uuid=rule_uuid,
+                    rule_run_at=rule_run_at,
+                    **action_args,
+                )
```

Comment:
@benthomasson do we have to send the result of the action_plugin to the AAP server, after it has finished execution. Will there be a post_event/set_fact at the end of this action. Would it make sense to extract out some of the logic from builtin which does the reporting for us.

Reply:
We probably should send the result.  Moving the reporting code out of builtin into a different module is a good idea.